src/Commands/Permission.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Althinect\FilamentSpatieRolesPermissions\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\File;
+use Spatie\Permission\Models\Permission as PermissionModel;
+use Illuminate\Support\Facades\DB;
+
+class Permission extends Command
+{
+    public $config;
+
+    public array $permissions = [];
+
+    protected $signature = 'permissions:sync {--C|clean}';
+
+    protected $description = 'Generates permissions through Models or Filament Resources and custom permissions';
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->config = config('filament-spatie-roles-permissions.generator');
+    }
+
+    public function handle(): void
+    {
+        $customModels = $this->getCustomModels();
+
+        $models = $this->getModels();
+
+        $classes = array_merge($customModels, $models);
+
+        $classes = array_diff($classes, $this->getExcludedModels());
+
+        $this->deleteExistingPermissions();
+
+        $this->prepareClassPermissions($classes);
+
+        $this->prepareCustomPermissions();
+
+        foreach ($this->permissions as $permission) {
+            $this->comment("Syncing Permission for: " . $permission['name']);
+            PermissionModel::firstOrCreate($permission);
+        }
+    }
+
+    public function deleteExistingPermissions(): void
+    {
+        if ($this->option('clean')) {
+            $this->confirm('This will delete existing permissions. Do you want to continue?', false);
+            $this->comment('Deleting Permissions');
+            DB::table(config('permission.table_names.permissions'))->delete();
+            $this->comment('Deleted Permissions');
+        }
+    }
+
+    public function prepareClassPermissions($classes): void
+    {
+        foreach ($classes as $class) {
+            foreach ($this->modelPermissions() as $modelPermission) {
+                foreach ($this->guardNames() as $guardName) {
+                    $this->permissions[] = [
+                        'name' => $class . '.' . $modelPermission,
+                        'guard_name' => $guardName
+                    ];
+                }
+            }
+        }
+    }
+
+    public function prepareCustomPermissions(): void
+    {
+        foreach ($this->getCustomPermissions() as $customPermission) {
+            foreach ($this->guardNames() as $guardName) {
+                $this->permissions[] = [
+                    'name' => $customPermission,
+                    'guard_name' => $guardName
+                ];
+            }
+
+        }
+    }
+
+    public function getModels(): array
+    {
+        $models = [];
+
+        if ($this->config['discover_models_through_filament_resources']) {
+            $resources = File::files(app_path('../app/Filament/Resources'));
```

Comment:
a try-catch could help here. e.g., resource not created yet.

Reply:
This retrieves all the resources that goes into the filament folder. So permissions will be generated only for resources that have been created.